docker-compose.yml

```diff
@@ -34,6 +34,25 @@ services:
       retries: 5
     restart: unless-stopped
 
+  email:
+    image: python_email:latest
+    build:
+      context: .
+      dockerfile: Dockerfile.email 
+    restart: always
+    volumes:
+      - email_data:/data 
+    environment: 
+      - PYTHONUNBUFFERED=1
+    ports:
+      - "1025:1025"
+    healthcheck:
+      test: ["CMD","netstat","-l","|","grep","1025"]
+      interval: 10s
+      timeout: 5s
+      retries: 5
+    command: python -m smtpd -d -n -c DebuggingServer 0.0.0.0:1025 
```

Comment:
I tried to manually send an email from the `coldfront` container, but got an error:

```
(app-root) bash-4.2# python manage.py shell
Python 3.8.6 (default, Oct 27 2020, 09:13:12) 
[GCC 9.3.1 20200408 (Red Hat 9.3.1-2)] on linux
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from coldfront.core.utils.mail import send_email
>>> send_email("test", "test", "meli@lbl.gov", ["meli@lbl.gov"])
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "/vagrant/coldfront_app/coldfront/coldfront/core/utils/mail.py", line 52, in send_email
    email.send(fail_silently=False)
  File "/opt/app-root/lib64/python3.8/site-packages/django/core/mail/message.py", line 284, in send
    return self.get_connection(fail_silently).send_messages([self])
  File "/opt/app-root/lib64/python3.8/site-packages/django/core/mail/backends/smtp.py", line 102, in send_messages
    new_conn_created = self.open()
  File "/opt/app-root/lib64/python3.8/site-packages/django/core/mail/backends/smtp.py", line 62, in open
    self.connection = self.connection_class(self.host, self.port, **connection_params)
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/smtplib.py", line 253, in __init__
    (code, msg) = self.connect(host, port)
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/smtplib.py", line 339, in connect
    self.sock = self._get_socket(host, port, self.timeout)
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/smtplib.py", line 308, in _get_socket
    return socket.create_connection((host, port), timeout,
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/socket.py", line 808, in create_connection
    raise err
  File "/opt/rh/rh-python38/root/usr/lib64/python3.8/socket.py", line 796, in create_connection
    sock.connect(sa)
OSError: [Errno 99] Cannot assign requested address
```

I think we need to tell Django to refer to the `email` container as the host for emails, rather than `localhost` (like we have `db_host` and `redis_host`, we may need `email_host`).

=====

You mentioned that you were able to make some requests that should send emails, but that logic is designed to not fail even if email-sending fails.

Reply:
I went and manually edited `local_settings.py` and changed the `EMAIL_HOST` variable to 'email' and I could run the django command above. The sample email displayed in the `std out` of in the Terminal window where I brought up the `docker compose`:

```
coldfront-email-1      | Incoming connection from ('172.18.0.5', 35920)
coldfront-email-1      | Peer: ('172.18.0.5', 35920)
coldfront-email-1      | Data: b'ehlo 8d324fb822fd'
coldfront-email-1      | Data: b'mail FROM:<dummy_email.com>'
coldfront-email-1      | ===> MAIL FROM:<dummy_email.com>
coldfront-email-1      | sender: dummy_email.com
coldfront-email-1      | Data: b'rcpt TO:<dev1@localhost>'
coldfront-email-1      | ===> RCPT TO:<dev1@localhost>
coldfront-email-1      | recips: ['dev1@localhost']
coldfront-email-1      | Data: b'rcpt TO:<dev2@localhost>'
coldfront-email-1      | ===> RCPT TO:<dev2@localhost>
coldfront-email-1      | recips: ['dev1@localhost', 'dev2@localhost']
coldfront-email-1      | Data: b'data'
coldfront-email-1      | Data: b'Content-Type: text/plain; charset="utf-8"\r\nMIME-Version: 1.0\r\nContent-Transfer-Encoding: 7bit\r\nSubject: [MyLRC-User-Portal] test\r\nFrom: dummy_email.com\r\nTo: dev1@localhost, dev2@localhost\r\nDate: Fri, 12 May 2023 19:32:33 -0000\r\nMessage-ID: <168391995368.193.5574160057182051133@8d324fb822fd>\r\n\r\ntest'
coldfront-email-1      | ---------- MESSAGE FOLLOWS ----------
coldfront-email-1      | b'Content-Type: text/plain; charset="utf-8"'
coldfront-email-1      | b'MIME-Version: 1.0'
coldfront-email-1      | b'Content-Transfer-Encoding: 7bit'
coldfront-email-1      | b'Subject: [MyLRC-User-Portal] test'
coldfront-email-1      | b'From: dummy_email.com'
coldfront-email-1      | b'To: dev1@localhost, dev2@localhost'
coldfront-email-1      | b'Date: Fri, 12 May 2023 19:32:33 -0000'
coldfront-email-1      | b'Message-ID: <168391995368.193.5574160057182051133@8d324fb822fd>'
coldfront-email-1      | b'X-Peer: 172.18.0.5'
coldfront-email-1      | b''
coldfront-email-1      | b'test'
coldfront-email-1      | ------------ END MESSAGE ------------
coldfront-email-1      | Data: b'quit'
```


How do we want to set this up? Should it be an instruction to edit the local_settings.py manually or should it be done in a setup script?

Reply:
`local_settings.py` is copied from `local_settings.py.sample`, so we could update that, but that file shouldn't change from instance to instance.

Instead, an instance-specific file is generated by Ansible (or `gen_config.sh`). Variables from `main.yml` are used to populate `settings_template.tmpl`. You can set `EMAIL_HOST` (maybe before `EMAIL_PORT`) in that file to the value of a new Ansible variable.

(Also, given that emails are written to `stdout`, sounds like we don't need the volume then.)

Reply:
Made the changes requested, `email` container is now recognized by `coldfront`.